src/Compiler/NodeVisitor/ChangeFunctionsNodeVisitor.php

```diff
@@ -284,4 +293,30 @@ private function createFunctionCallNode(string $functionName, array $args): ?Nod
             $args
         );
     }
+
+    /**
+     * @param BetterReflectionFunction|BetterReflectionMethod|ReflectionFunction $reflection
+     * @param Arg[]|VariadicPlaceholder[] $args
+     * @return Arg[]|VariadicPlaceholder[]
+     */
+    private function updateArgs($reflection, array $args): array
```

Comment:
Remove first argument if we pass $this and function does not expect Tempalte as first argument. New latte always passes this and FunctionExecutor handles this logic